src/map/skill.c

```diff
@@ -11709,6 +11863,103 @@ static int skill_castend_nodamage_id(struct block_list *src, struct block_list *
 	return 0;
 }
 
+/**
+ * Castend handler for Song/Dance skills that simply cause SCs (RE-only)
+ * @param src Unit who cast the skill
+ * @param bl Unit being targeted by the skill
+ * @param skill_id skill being cast
+ * @param skill_lv level of the skill being cast
+ * @param tick
+ * @param flag castend flags
+ * - flag & 1 :
+ *     when not set: it is the initial skill cast,
+ *     when set: it is in the reiteration over targets to get buffed.
+ * - enum e_battle_check_target flags: defines who will be affected by the reiteration. examples:
+ *     flag | BCT_PARTY -- affects party members (e.g. apply SC to all party members)
+ *     flag | BCT_ENEMY -- affects enemies (e.g. apply SC to all enemies)
+ *
+ * - @TODO: Document other possible flag values
+ */
+static void skill_castend_nodamage_id_sc_song(struct block_list *src, struct block_list *bl, uint16 skill_id, uint16 skill_lv, int64 tick, int flag)
```

Comment:
I tried to start making the idea of "separate function for each skill" a thing here, since most dance/songs are very similar. But I ended up with:
1. this one having a if for lullaby
2. `skill_castend_nodamage_id_ugly_dance` having a duplicated code for most of it
3. the caller function needing to give the target

I am open for suggestions on how I could improve that, I had a few random thoughts but didn't tried too much:
1. Moving chance to SkillData2 (but except for 1 case, it is always 100%...)
2. Making more functions, with their exclusive logic and those would call this one. Something like:
     - song_friend_sc --> sc_song w/ Party + chance = 100%
     - song_enemy_sc --> sc_song w/ enemy + chance = 100%
     - song_lullaby --> sc_song w/ enemy + chance = 15%
     - song_ugly_dance --> sc_song w/ enemy + special logic for flag 1

But well, I am not sure whether those would work nicely.